huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go

```diff
@@ -1749,26 +1703,59 @@ func deleteFunctionVersionAlias(client *golangsdk.ServiceClient, functionUrn, al
 	return nil
 }
 
-func deleteFunctionVersions(client *golangsdk.ServiceClient, functionUrn string, versions []interface{}) error {
-	// In the future, the function will support manage multiple versions.
-	for _, version := range versions {
-		versionNum := utils.PathSearch("name", version, "").(string) // The version name, also name as the version number.
-		if versionNum != "latest" {
-			// Deletes a function version, also deleting all aliases beneath it.
-			err := deleteFunctionOrVersion(client, fmt.Sprintf("%s:%s", functionUrn, versionNum))
-			if err != nil {
-				return err
-			}
-			continue
-		}
-		// Since the latest version cannot be deleted, only the version alias under it can be deleted.
-		aliases := utils.PathSearch("aliases", version, make([]interface{}, 0)).([]interface{})
-		for _, alias := range aliases {
-			err := deleteFunctionVersionAlias(client, functionUrn, utils.PathSearch("name", alias, "").(string))
-			if err != nil {
-				return err
-			}
-		}
+func createFunctionVersion(client *golangsdk.ServiceClient, functionUrn, versionName, versionDesc string) error {
+	httpUrl := "v2/{project_id}/fgs/functions/{function_urn}/versions"
+
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createPath = strings.ReplaceAll(createPath, "{function_urn}", functionUrn)
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+		JSONBody: map[string]interface{}{
+			"version":     versionName,
+			"description": versionDesc,
+		},
+	}
+
+	_, err := client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return fmt.Errorf("failed to create the function version: %s", err)
+	}
+	return nil
+}
+
+func createFunctionVersionAlias(client *golangsdk.ServiceClient, functionUrn, versionName string, aliasCfg interface{}) error {
+	if aliasCfg == nil {
+		return nil
+	}
+
+	httpUrl := "v2/{project_id}/fgs/functions/{function_urn}/aliases"
+
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createPath = strings.ReplaceAll(createPath, "{function_urn}", functionUrn)
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+		JSONBody: map[string]interface{}{
+			"version":     versionName,
+			"name":        utils.ValueIgnoreEmpty(utils.PathSearch("name", aliasCfg, nil)),
+			"description": utils.ValueIgnoreEmpty(utils.PathSearch("description", aliasCfg, nil)),
+			"additional_version_weights": utils.ValueIgnoreEmpty(utils.StringToJson(utils.PathSearch("additional_version_weights",
+				aliasCfg, "").(string))),
+			"additional_version_strategy": utils.ValueIgnoreEmpty(utils.StringToJson(utils.PathSearch("additional_version_strategy",
```

Comment:
The function `ValueIgnoreEmpty` may return nil value.
Does it make sense to send nil value?

Reply:
sure about that, the utils.RemoveNil() method is used.